lib/index.js

```diff
@@ -974,3 +984,23 @@ exports.shallow = function (source) {
 
     return target;
 };
+
+
+exports.promiseWrap = function (bind, method, args, transform) {
+
+    return new Promise((resolve, reject) => {
+
+        const callback = function (error, result) {
+
+            if (Array.isArray(transform)) {
+                result = exports.transform(Array.from(arguments), transform);
+                error instanceof Error ? reject(result) : resolve(result);
```

Comment:
indeed, but will make it harder to allow current projects to support promises, and imo this what this method is meant to do , if we limit it, we may end up with it un-used, if we don't limit we may end up with poorly design stuff ... tough decision